### PR TITLE
Test convertability of AD-regular Rank*Tensors

### PR DESCRIPTION
**Pull request by lindsayad**
It was revealed by Ferret that we didn't have a proper
assignment operator for RankThreeTensors for different
template parameters. Now we test convertability between
all the different Rank*Tensors in MOOSE

Refs idaholab/moose#14701

### CODE REVIEW

**Review by fdkong — DISMISSED** [review]
@lindsayad Could you make sure the PR is able to pass `precheck` before you merge?

File: test/tests/kernels/simple_diffusion/simple_diffusion.i [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [Mesh]
   type = GeneratedMesh
-  dim = 2
-  nx = 10
-  ny = 10
+  dim = 1
+  nx = 2
+  # ny = 2
```
Comment (lindsayad) [inline test/tests/kernels/simple_diffusion/simple_diffusion.i:5]: damn it this is what I get for trying to be funny @fdkong

Comment (fdkong) [reply test/tests/kernels/simple_diffusion/simple_diffusion.i]: Haha

**Comment by lindsayad** [comment]
@mangerij I didn't actually need to modify ferret at all. I believe this PR should smooth things over

**Review by fdkong — APPROVED** [review]

**Comment by mangerij** [comment]
Great!
________________________________

**Pull request merged**